library-best-practices/JLBP-17.md

```diff
@@ -0,0 +1,38 @@
+[JLBP-17] Coordinate Major Version Adoption
+-------------------------------------------
+
+It is clearly undesirable for a dependency tree to contain conflicts.
+Unfortunately, as soon as any library in a dependency tree introduces a breaking
+change, consumers can't use the latest versions of those libraries in the
+dependency tree together. To enable clients to depend on the latest versions of
+their dependencies, breaking changes should be propagated as quickly as
+ possible.
+
+Perform the rollout in this manner:
+
+1. Decide whether to break the feature as an atomic change or to perform the
+   change in two passes, deprecate then delete.
+   - If the feature is used by stable code in other libraries, you must use two
+     passes (as per [JLBP-7](JLBP-7.md)).
+   - If the code does not appear to be used in other libraries, an in-place
```

Comment:
Some breaking changes may not be able to be done with a deprecation first. For example, changing a class to final or removing or adding a checked exception in a method's throws clause. Should we mention this?

Reply:
That makes sense for the case of making something `final`, but for the checked exception case, you could technically introduce a new method whose difference is the exceptions it declares to throw. It's worth mentioning this type of change, though.